plugins/utils/trimmer/copy.c

```diff
@@ -91,67 +94,43 @@ enum bt_component_status update_packet_context_field(FILE *err,
 	struct bt_ctf_field *field = NULL, *writer_field = NULL;
 	int nr_fields, i, int_ret;
 
+	BT_LOGI("Updating packet context field: name=%s", name);
 	packet_context = bt_ctf_packet_get_context(writer_packet);
-	if (!packet_context) {
-		fprintf(err, "[error] %s in %s:%d\n", __func__, __FILE__,
-				__LINE__);
-		goto error;
-	}
+	assert(packet_context);
 
 	struct_type = bt_ctf_field_get_type(packet_context);
-	if (!struct_type) {
-		fprintf(err, "[error] %s in %s:%d\n", __func__, __FILE__,
-				__LINE__);
-		goto error;
-	}
+	assert(struct_type);
 
 	writer_packet_context = bt_ctf_packet_get_context(writer_packet);
-	if (!writer_packet_context) {
-		fprintf(err, "[error] %s in %s:%d\n", __func__, __FILE__,
-				__LINE__);
-		goto error;
-	}
+	assert(writer_packet_context);
```

Comment:
Everything that starts with "writer_" is something we created and we can modify, the other is what we receive. At this stage, if the input has a packet_context, we have created a writer_packet_context.